boot/zephyr/include/mcuboot_config/mcuboot_config.h

```diff
@@ -185,6 +185,10 @@
 #define MCUBOOT_SERIAL_DIRECT_IMAGE_UPLOAD
 #endif
 
+#ifdef CONFIG_BOOT_SERIAL_WAIT_FOR_DFU
+#define MCUBOOT_SERIAL_WAIT_FOR_DFU
+#endif
+
```

Comment:
Make this something like:
```
#ifdef CONFIG_MCUBOOT_SERIAL_WAIT_FOR_DFU
#define MCUBOOT_SERIAL_WAIT_FOR_DFU CONFIG_MCUBOOT_SERIAL_WAIT_FOR_DFU
#else
#define MACUBOOT_SERIAL_WAIT_FOR_DFU 0
#endif
```
Add the definition, of `MCUBOOT_SERIAL_WAIT_FOR_DFU`, or each file system supported by Zephyr (just search mcuboot tree for them); for example (shameless plug) https://github.com/mcu-tools/mcuboot/blob/main/boot/mynewt/mcuboot_config/include/mcuboot_config/mcuboot_config.h#L89-L93

This way the code will not be available for systems that do not support it without `#ifdef __ZEPHYR__`, you just can add `#if MCUBOOT_SERIAL_WAIT_FOR_DFU != 0`.

Reply:
Thanks for the hint. however i'm looking into the different mcuboot_config.h, and doesn't look like it's common at all to define something to 0 if the feature isn't availlable like for the mynewt you've shown. I will stick with the ifdef instead of assigning a value since that would require all configs to be touched/polluted if you ask me. 
I'll do the changes, and you can argue with this comment if you don't like it, and we can see for alternative options then maybe?

Reply:
I do not think that this would pollute anything, you have to have separate commits for other sys though, this rather informs users that configure mcuboot for their systems that such option exists in common code but is not supported by their system yet.
We do not have common, generic, config file that would gather all common code defines and then a system specific code would override them, so this is way to inform users about possibilities.

Reply:
@d3zd3z Sorry for dragging you here, but I would like your opinion on the approach as I do not want to talk @wcappelle into doing something that mcuboot does not care about.